cmake/ext_ortlib.cmake

```diff
@@ -14,21 +14,23 @@ else()
   if(CMAKE_HOST_APPLE)
     set(ONNXRUNTIME_URL "v${ONNXRUNTIME_VER}/onnxruntime-osx-universal2-${ONNXRUNTIME_VER}.tgz")
   elseif(CMAKE_HOST_WIN32)
```

Comment:
Actually, here you should not use the "HOST" word.

"Host" mean the environment that is used for running the compilers.
"Target" means the environment that will be used for running the built binaries. 

So, if your cross-compile a Win32 binary on a Linux machine, your host environment is Linux, your target environment is Windows. And you want to link your application to Windows libraries instead of Linux libraries. 

That's why in ORT's code usually we use [WIN32](https://cmake.org/cmake/help/latest/variable/WIN32.html) instead of CMAKE_HOST_WIN32. 

See https://github.com/microsoft/onnxruntime/blob/main/docs/cmake_guideline.md#basics-of-cross-compiling for more information about cross-compiling. 

You don't have to support cross-compiling, but it's better to use the right macros.

Reply:
FWIW this is used to select a pre-built ORT binary for use in C++ unit tests. Currently the setup doesn't support running those when cross-compiling so we wouldn't actually use the ORT library selected in that case. 

We have some separate Android and iOS testing but that uses the published ORT packages directly when building the test app (and it's a test app rather than a unit test). 

I can update to use WIN32/APPLE to be more correct though